src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginUtil.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using System.Collections.Generic;
+
+namespace Agent.Plugins.PipelineArtifact
+{
+    public static class PipelineArtifactPathHelper
```

Comment:
>PipelineArtifactPathHelper [](start = 24, length = 26)

This isn't used to validate paths, it's used to validate artifact names. Change the class name to PipelineArtifactNameValidator

Reply:
The whole point of validating the artifact name is because it will be part of the file path right? We are not validating artifact name because they are artifact names. The check is really checking the right file names and I think it's more logically correct to name it PipelineArtifactPathHelper.

Reply:
The usage is confusing:
if(!PipelineArtifactPathHelper.IsValidPath(artifactName)) {
                throw new ArgumentException(StringUtil.Loc("ArtifactNameIsNotValid"));
            }

artifactName is being passed as the path but the artifactName isn't a path in this game, it's just an arbitrary string

---
In reply to: [291394782](https://github.com/microsoft/azure-pipelines-agent/pull/2279#discussion_r291394782) [](ancestors = 291394782)

Reply:
Either pass the full path which this will be fine, it will restrict the path chars or you need to change the usage to just the artifact name.

---
In reply to: [291395223](https://github.com/microsoft/azure-pipelines-agent/pull/2279#discussion_r291395223) [](ancestors = 291395223,291394782)

Reply:
Actually I'm in favor of changing to just restrict it to artifactName because this is not accurate. '/', '\\' are valid path characters in NTFS, not valid file name characters. This is expressing that the path cannot contain these characters, but it's just restricted to files so the language isn't exactly correct.

---
In reply to: [291395416](https://github.com/microsoft/azure-pipelines-agent/pull/2279#discussion_r291395416) [](ancestors = 291395416,291395223,291394782)

Reply:
To be precise these are invalid characters for "file names"

Reply:
you can use Path.GetInvalidFileNameChars then

Reply:
no need to maintain your own set of chars

Reply:
Ah we cannot use that for one reason and that's because that would OS specific, in this case we need it standardized across OSes so I would change the method name and put a comment as to why we cannot use that.

Reply:
Disagree. FCS is using the same way to do it. And from looking at the Path.GetInvalidFileNameChar description - "The array returned from this method is not guaranteed to contain the complete set of characters that are invalid in file and directory names. The full set of invalid characters can vary by file system. "

Reply:
Right so I would change the name because the usage is not accurate

Reply:
check comment above

Reply:
I would change "path" to fileName. but for the class name, it's more extensible to name it PipelineArtifactPathHelper not just fileNameHelper?

Reply:
It makes it sound like any files in pipeline artifacts cannot have those characters which is not true. We're really restricting those characters to just the artifact name, but some of those characters are allowed depending on the OS.

Reply:
I think it's the other way around. And we have almost the name naming like FCS. Here we are implying all artifact name should follow this rules but that's not true. The only reason we are doing it is because it'll be a file name

Reply:
you mean directory not file name

Reply:
All artifact names under pipeline artifacts should follow this rule because the class is prefixed with  "PipelineArtifact<whatever>", it doesn't apply to FCS

Reply:
This apply to both file or folder names

Reply:
Ok I think this is really confusing because when I think of PipelineArtifactPathHelper, I think of it validating the paths of the contents. And whatever it is doing is not doing that right now and the logic is not correct for that usage because technically "/" and "\" are allowed in the paths. That's why I'm proposing to change it to PipelineArtifactNameValidator because we're restricting the these characters to the artifact name, not the contents.

Reply:
Basically these chars - / \ : * ? " < > | are not allowed as a file/folder names in NTFS.

Reply:
We can't just work off of a NTFS basis, Pipeline Artifacts also works in macOS + Linux so we cannot just say that * ? " < > | are disallowed in any path.